tce/explorer/BlocksConnectIC.cc

```diff
@@ -0,0 +1,266 @@
+/*
+    Copyright (c) 2002-2021 Tampere University.
+
+    This file is part of TTA-Based Codesign Environment (TCE).
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+ */
+/**
+ * @file BlocksConnectIC.cc (Derrived from VLIWConnectIC.cc)
+ *
+ * Explorer plugin that generates connections for Blocks machine
+ *
+ * @author Kanishkan Vadivel 2021
+ * @note rating: red
+ */
+
+#include <algorithm>
+#include <vector>
+
+#include "ControlUnit.hh"
+#include "Conversion.hh"
+#include "DSDBManager.hh"
+#include "DesignSpaceExplorerPlugin.hh"
+#include "Exception.hh"
+#include "Guard.hh"
+#include "HDBRegistry.hh"
+#include "Machine.hh"
+#include "Segment.hh"
+
+// using namespace TTAProgram;
+using namespace TTAMachine;
+using namespace HDB;
+using std::endl;
+
+class BlocksConnectIC : public DesignSpaceExplorerPlugin {
+    PLUGIN_DESCRIPTION(
+        "Arranges architecture FUs into a Blocks-like interconnection");
+
+    BlocksConnectIC() : DesignSpaceExplorerPlugin() {}
+
+    virtual bool
+    requiresStartingPointArchitecture() const {
+        return true;
+    }
+    virtual bool
+    producesArchitecture() const {
+        return true;
+    }
+    virtual bool
+    requiresHDB() const {
+        return false;
+    }
+    virtual bool
+    requiresSimulationData() const {
+        return false;
+    }
+    virtual bool
+    requiresApplication() const {
+        return false;
+    }
+
+    /**
+     * Arranges architecture FUs into a Blocks-like interconnection. This is
```

Comment:
Brief comment, separate by empty line. Perhaps add reference URL for Blocks?

Reply:
Added Mark's thesis!